### ISSUE TITLE
Feature request: Special Functions -> Play sound

### ISSUE THREAD

**Issue by MrD-RC** [FrSkyRC/ETHOS-Feedback-Community#661]
- [x] Reorganize the audio folder
       - system - en
                      - [language]
- [x] Allow to select a Subdirectory, then a File in PlayTrack 
       So that the user can have such directories
       - sounds - beep.wav
                      - ding.wav

----

It would be handy to have the Play Sound option in the special functions. The could be a collection of beeps and generic sounds that we could apply to switch conditions.

I understand that we could create them ourselves and put them in the audio folder. But this approach doesn’t lend its self to ease of use for people new to ETHOS or flying in general. 

However, what may be useful is to have a **Sounds** folder on the MicroSD card, that people can add their own to. This could be located in the **Audio** folder. I think it would be good to keep the files for spoken and sounds separate.

**Comment by Hobby4life** [comment]
I would like to link to this too: #442

**Comment by bsongis-frsky** [comment]
You may add all your sounds in the Audio folder

**Comment by MrD-RC** [comment]
@bsongis-frsky true. But that leads to a big list and disorganisation. It would be better to have separate sound effects (OpenTX sounds) and voice commands and feedback (OpenTX tracks)

If this is not something that you want to put in the firmware. At least consider adding the subfolders of the Audio folder in the selection list. That way the RC pilots can organise the audio files as they see fit, and make things easier to find. 

I know with the current folder layout, that would show all the language folders. But, considering that all language folders now only contain system tracks. Why not reverse the folder order from lang/system to system/lang. That way you wouldn’t have a lot of subfolders showing for languages. You could even hide the single system folder if you wish.

**Comment by bsongis-frsky** [comment]
Now it makes sense, I will update the issue accordingly

**Comment by Hobby4life** [comment]
Language bound Soundpacks should literally have their own folder structure.

Placing your own extra sound files only in the Audio folder is not very practical.

Why...

I occasionally fly into Germany, so I use a German soundpack there. I can switch that in the system menu. OK.
But to convert all my own associated sound files to German, I literally have to replace everything in the audio folder on the PC. I can't do this on the transmitter.

Now when I'm back in NL (I prefer the English soundpack), then I have to restore everything in the audio folder via the PC.

That's just totally inconvenient.

Everything that belongs to a language should be in that specific language folder and not outside it.

As an extra, it would be great if I could choose between, for example, Various English soundpacks.

**Comment by Farabos** [comment]
How can i stop a playing track ?
In OpenTX there was a special function for this.

**Comment by azaz44** [comment]
I'm also missing all the beeps, dings, warnings and chirps from OpenTX.

**Comment by Target0815** [comment]
> The new facility for voices allows up to 3 sets of sounds to be selected, as of 1.5.0 nightly

What does the directory structure for this look like? I've read about it in the draft manual and I'm wondering who is supposed to keep track of it.

I would like to have the German language and additionally "Schwäbisch" and "Plattdütsch" (just like that with the umlauts ...). How would that look in the directory structure?

**Comment by MIKE-DAILEY** [comment]
There is a "how to" post here on RCG https://www.rcgroups.com/forums/showpost.php?p=50718883&postcount=23264

**Comment by Target0815** [comment]
> There is a "how to" post here on RCG https://www.rcgroups.com/forums/showpost.php?p=50718883&postcount=23264

Yes, I saw that post. But who comes up with such directory structures? That is completely illogical ...

An "audio" directory is required. Furthermore, a subdivision into "system" and "user" files. And **only then** a subdivision by language ...

It should look like this:

audio/system/en (us, de, etc.)
audio/user1/en
audio/user2/en
audio/user3/en

Here you would have a clear structure and the desired 3 user voices. This is easy to understand and even easier to program.

Please dispose of the other rubbish ... that was already a cramp in 1.4.x.

**Comment by bsongis-frsky** [comment]
The "other rubbish" allows to have a different voice also on system prompts

**Comment by bsongis-frsky** [comment]
Who could help working on a nice "Sound pack" with all sort of ding, beep, etc.

**Comment by Target0815** [comment]
> The "other rubbish" allows to have a different voice also on system prompts

That may be the case, but IMHO it is not relevant because the system and user language are almost always set to the same settings. It is much more relevant that there is no clear structure. This confuses users and ultimately files end up in the wrong directories and that generates additional support requests. I can read about this every day in the German forum ...

We already had a similar issue with the LUAs for ELRS, Crossfire, MPM etc. back then. At that time you changed your mind and it is still running smoothly today.

Please think about it again.

**Comment by Target0815** [comment]
@mawzthefinn Can you show me one of these complaints?

**Comment by mawzthefinn** [comment]
Check the X20 thread on RCGroups.

**Comment by mawzthefinn** [comment]
Closing this off as the original request has been addressed in ETHOS 1.5

**Comment by azaz44** [comment]
> Closing this off as the original request has been addressed in ETHOS 1.5

Was this really solved? Organizarion of sound packs into folders is better now, but do we have beeps and sounds other then speech?

**Comment by mawzthefinn** [comment]
> > Closing this off as the original request has been addressed in ETHOS 1.5
> 
> Was this really solved? Organizarion of sound packs into folders is better now, but do we have beeps and sounds other then speech?

Beeps and other sounds have an issue of their own #1281

**Comment by Jedsters** [comment]
I notice #1281 which references this issue has had enhancement removed from it and is being allowed to go stale while this issue which references #1281 has been closed. It therefore appears that there is no plan to address this issue or #1281. 
I could remove the stale but sometimes I get fed up trying to improve things when it appears there is no interest for some issues. 

I've added my own customs sounds to my tx and as in #1281 have provided those sounds. So to use an English expression, I'm all right jack.
